frontend/src/app/shared/services/auth/auth.service.ts

```diff
@@ -0,0 +1,68 @@
+import {Injectable} from "@angular/core";
+import {Config} from "../../../config/Config";
+import {tokenNotExpired} from "angular2-jwt";
+import {NotificationsService} from "angular2-notifications";
+import {TranslateService} from "ng2-translate";
+import {Router} from "@angular/router";
+
+declare var Auth0Lock: any;
+
+@Injectable()
+export class AuthService {
+    private auth0Config: any;
+    private readonly supportedLangs: Array<string> = ['de', 'en'];
+    lock;
+
+    constructor(private config: Config, private router: Router, private translateService: TranslateService) {
+        this.auth0Config = this.config.get('auth0');
+
+        this.lock = new Auth0Lock(this.auth0Config.clientID, this.auth0Config.domain, {
+            auth: {
+                redirectUrl: this.auth0Config.callbackURL,
+                responseType: 'token'
+            },
+            rememberLastLogin: false,
+            allowForgotPassword: false,
+            loginAfterSignUp: true,
+            additionalSignUpFields: [{
+                name: 'name',
+                placeholder: 'Enter your username' // todo i18n
+            }],
+            theme: {
+                logo: 'https://avatars0.githubusercontent.com/u/22800787?v=3&s=200',
```

Comment:
I think we should host this ourselves. I put it here https://cdn.peg.nu/sites/outcobra/logo.jpg where it is globally distributed by KeyCDN.